### PR TITLE
caching: Local Locking fix

### PR DESCRIPTION
**Pull request by kujenga**
Tests have been added for an issue with the new `LocalLock` functionality in PR https://github.com/rastech/gometa/pull/12 where the lock only applied to the cacher instance itself, which doesn't match with how these cachers are used.

The added tests uncovered an issue with the `Context` cacher which will be resolved shortly.

### CODE REVIEW

**Comment by AndyHaskell** [comment]
So just to make sure we're on the same page, the bug was that because the mut/mutmap are per-Memcache, that would only be shared btwn that one Memcache, rather than all of them right?

**Comment by kujenga** [comment]
Yep! That's correct. And since instances of a Cacher are generally treated as transient, it basically rendered the `LocalLock` a no-op.

File: caching/memcache_test.go [hunk 1]
```diff
@@ -10,6 +10,10 @@ func (s *cachingSuite) TestMemcacheParGetMulti() {
 	testParGetMulti(s.T(), 3, 3, 1, &Memcache{Context: s, Expiration: 30 * time.Second})
 }
 
+func (s *cachingSuite) TestMemcacheMultiCacherGet() {
+	testMultiCacherGet(s.T(), 10, 1, func() Cacher { return &Memcache{Context: s, Expiration: 30 * time.Second} })
```
Comment (AndyHaskell) [inline caching/memcache_test.go]: Do we want `LocalLock here?`

Comment (kujenga) [inline caching/memcache_test.go]: I don't think so, there's a separate test (`TestMemcacheMultiCacherGetLocalLock`) for that below. Why do you ask?

Comment (AndyHaskell) [inline caching/memcache_test.go]: Oh missed that

**Comment by AndyHaskell** [comment]
:+1:

**Pull request merged**